Test1/src/leetcode/LeetCode40.java

```diff
@@ -0,0 +1,47 @@
+package leetcode;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * https://leetcode.com/problems/combination-sum-ii/
+ * @author minibig
+ *
+ */
+public class LeetCode40 {
+	
+	int candidatesSize = 100;
+
+	public List<List<Integer>> combinationSum2(int[] candidates, int target) {
+		
+		List<List<Integer>> answerList = new ArrayList<>();
+		
+		int[] candidatesSort = new int[candidatesSize + 1];
+		
+		for(int num : candidates) {
+			candidatesSort[num]++;
```

Comment:
I'm personally against counting sort in almost any cases.

If you use counting sort, the space complexity will be dependent upon the maximum number, and this is not ideal.

I think a standard O(NlogN) sorting method would be enough in this case.